packages/react-native-sdk/src/contexts/ThemeContext.tsx

```diff
@@ -29,11 +29,10 @@ export type ThemeContextValue = {
 
 export const mergeThemes = (params: MergedThemesParams) => {
   const { style, theme } = params;
-  const finalTheme = (
-    !theme || Object.keys(theme).length === 0
-      ? JSON.parse(JSON.stringify(defaultTheme))
```

Comment:
Had to switch to `Object.assign()` because the old solution was dropping the methods defined in the theme object. @khushal87 thanks again for helping me debug this. 🚀

Reply:
hey @kristian-mkd, keep in mind that `Object.assign` does a shallow merge. Perhaps you want to use `lodash/deepMerge`.

Reply:
I think I can use `lodash/deepClone` first and then `lodash/merge`, something like this:
```ts
  const finalTheme = isThemeEmpty ? cloneDeep(defaultTheme) : cloneDeep(theme);
  if (style) {
    merge(finalTheme, style);
  }
```

Reply:
this change is also necessary when you remove the method